cmd/internal/codegen/parse/parser.go

```diff
@@ -114,6 +120,37 @@ func (b *APIs) parseDomain() {
 	comments := Comments(pkg.Comments)
 	b.Domain = comments.getTag("domain", "=")
 	if len(b.Domain) == 0 {
-		panic("Could not find string matching // +domain=.+ in apis/doc.go")
+		b.Domain = parseDomainFromFiles(b.context.Inputs)
+		if len(b.Domain) == 0 {
+			panic("Could not find string matching // +domain=.+ in apis/doc.go")
+		}
+	}
+}
+
+func parseDomainFromFiles(paths []string) string {
+	var domain string
+    for _, path := range paths {
+    	if strings.HasSuffix(path, "pkg/apis") {
+    		filePath := strings.Join([]string{build.Default.GOPATH, "src", path, "doc.go"}, "/")
+            lines := []string{}
+
+			file, err := os.Open(filePath)
+			if err != nil {
+				glog.Fatal(err)
+			}
+			defer file.Close()
+			scanner := bufio.NewScanner(file)
+			for scanner.Scan() {
+				lines = append(lines, strings.Replace(scanner.Text(), "// ", "", 1))
+			}
+			if err := scanner.Err(); err != nil {
+				glog.Fatal(err)
+			}
+
+			comments := Comments(lines)
```

Comment:
May be, we can move this logic in the `scanner` loop itself itself. Ignore the line if its not a comment, else invoke Comments(line) and check for domain ?

Reply:
I updated this function to ignore the lines not a comment. 
`Comments(line)` is left after all comments are get since `Comments` is the same type of `[]string`.